portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,4 +50,20 @@ private String convertToJson(ArrayList list) {
     String json = gson.toJson(list);
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = request.getParameter("name-input");
+    String comment = request.getParameter("comment-input");
+    
+    ArrayList<String> tmp = new ArrayList<>();
+    tmp.add(name);
+    tmp.add(comment);
+
+    commentList.add(tmp);
```

Comment:
1. Avoid variable name like "tmp", name it to reflect the content for better readability, e.g. if you name the text message as commentMessage, you could name all the metadata with the message as a comment.

2. Define a separate class for comment to include both name and comment message instead of using a list of string: 

a) Using a list to include metadata will make it hard when you want to read from the list to decide which index is corresponding to which property. And it might also cause exception when you are adding new properties and by accident not adding property in the intended order, etc.

b) Wrapping into a class will make it more extendable to add more properties (e.g. timestamp) to support new features. Also it will provide you with the flexibility to store not only the string type but also other possible values like numbers. It will also make the code more readable.

For 2, you might figure out in step-5-datastore-walkthrough where you will need a class anyway, thus I am not blocking this pr. Just FYI, you might want to get to step-5 with a wrapper class to make it easier for features that needs more properties besides the text messages.

Reply:
I committed a change which defines a separate class for comment.